api/resources_portal/views/document_views.py

```diff
@@ -0,0 +1,330 @@
+            ),
+            openapi.Parameter(
+                name="organism",
+                in_=openapi.IN_QUERY,
+                type=openapi.TYPE_STRING,
+                description="Allows filtering the results by organism",
+            ),
+            openapi.Parameter(
+                name="has_publication",
+                in_=openapi.IN_QUERY,
+                type=openapi.TYPE_STRING,
+                description="Allows filtering the results by has_publication",
+            ),
+            openapi.Parameter(
+                name="has_pre_print",
+                in_=openapi.IN_QUERY,
+                type=openapi.TYPE_STRING,
+                description="Allows filtering the results by has_pre_print",
+            ),
+        ],
+        operation_description="""
+Use this endpoint to search among the experiments.
+
+This is powered by ElasticSearch, information regarding advanced usages of the
+filters can be found in the [Django-ES-DSL-DRF docs](https://django-elasticsearch-dsl-drf.readthedocs.io/en/0.17.1/filtering_usage_examples.html#filtering)
+
+There's an additional field in the response named `facets` that contain stats on the number of results per filter type.
+
+Example Requests:
+```
+?search=zebrafish
+?id=1
+?search=zebrafish&category=plasmid&has_publication=true
+?ordering=source_first_published
+```
+""",
+    ),
+)
+class MaterialDocumentView(DocumentViewSet):
+    """Defines search and filter fields for Materials"""
+
+    document = MaterialDocument
+    serializer_class = MaterialDocumentSerializer
+    pagination_class = ESLimitOffsetPagination
+
+    # Filter backends provide different functionality we want
+    filter_backends = [
+        FilteringFilterBackend,
+        OrderingFilterBackend,
+        DefaultOrderingFilterBackend,
+        CompoundSearchFilterBackend,
+        FacetedSearchFilterBackendExtended,
+    ]
+    lookup_field = "id"
+
+    # Define search fields
+    # Is this exhaustive enough?
+    search_fields = {
+        "title": {"boost": 10},
+        "category": None,
+        "additional_metadata": None,
+        "pubmed_id": None,
+        "organism": None,
+    }
+
+    # Define filtering fields
+    filter_fields = {
+        "id": {"field": "_id", "lookups": [LOOKUP_FILTER_RANGE, LOOKUP_QUERY_IN],},
+        "category": "category",
+        "organization": {"field": "name"},
+        "title": "title",
+        "organism": "organism",
+        "has_publication": "has_publication",
+        "has_pre_print": "has_pre_print",
+    }
+
+    # Define ordering fields
+    ordering_fields = {
+        "id": "id",
+        "title": "title.raw",
+        "category": "category.raw",
+        "created_at": "created_at",
+        "updated_at": "updated_at",
+    }
+
+    # Specify default ordering
+    ordering = (
+        "_score",
+        "created_at",
```

Comment:
@dvenprasad should materials be sorted by created_at or updated_at by default?

Reply:
We should do updated_at

Reply:
Ok, and that's true for every search route?

Reply:
I changed all created_at instances to updated_at, on the material, user, and organization routes.